go/dpagg/mean_test.go

```diff
@@ -738,3 +738,153 @@ func TestBMFloat64Serialization(t *testing.T) {
+			meanOpt:      &BoundedMeanFloat64Options{Lower: 0.2, Upper: 15, Noise: noNoise, MaxContributionsPerPartition: 1},
+			sumConfInt:   noise.ConfidenceInterval{LowerBound: 5, UpperBound: 10},
+			countConfInt: noise.ConfidenceInterval{LowerBound: 5, UpperBound: 10},
+			want:         noise.ConfidenceInterval{LowerBound: 8.1, UpperBound: 9.6},
+		},
+		{ // Negative lower and upper bounds for sum.
+			meanOpt:      &BoundedMeanFloat64Options{Lower: 1, Upper: 7, Noise: noNoise, MaxContributionsPerPartition: 1},
+			sumConfInt:   noise.ConfidenceInterval{LowerBound: -15, UpperBound: -5},
+			countConfInt: noise.ConfidenceInterval{LowerBound: 2, UpperBound: 4},
+			want:         noise.ConfidenceInterval{LowerBound: 1, UpperBound: 2.75},
+		},
+		{ // Negative lower bound for sum.
+			meanOpt:      &BoundedMeanFloat64Options{Lower: 1, Upper: 7, Noise: noNoise, MaxContributionsPerPartition: 1},
+			sumConfInt:   noise.ConfidenceInterval{LowerBound: -10, UpperBound: 0},
+			countConfInt: noise.ConfidenceInterval{LowerBound: 2, UpperBound: 4},
+			want:         noise.ConfidenceInterval{LowerBound: 1, UpperBound: 4},
+		},
+		// Clamp too low bounds.
+		{
+			meanOpt:      &BoundedMeanFloat64Options{Lower: 1, Upper: 9, Noise: noNoise, MaxContributionsPerPartition: 1},
+			sumConfInt:   noise.ConfidenceInterval{LowerBound: -100, UpperBound: -50},
+			countConfInt: noise.ConfidenceInterval{LowerBound: 2, UpperBound: 5},
+			want:         noise.ConfidenceInterval{LowerBound: 1, UpperBound: 1},
+		},
+		// Clamp too high bounds.
+		{
+			meanOpt:      &BoundedMeanFloat64Options{Lower: 1, Upper: 9, Noise: noNoise, MaxContributionsPerPartition: 1},
+			sumConfInt:   noise.ConfidenceInterval{LowerBound: 70, UpperBound: 100},
+			countConfInt: noise.ConfidenceInterval{LowerBound: 2, UpperBound: 5},
+			want:         noise.ConfidenceInterval{LowerBound: 9, UpperBound: 9},
+		},
+	} {
+		mean := NewBoundedMeanFloat64(tc.meanOpt)
+		mean.normalizedSum.noise = getMockConfInt(tc.sumConfInt)
+		mean.count.noise = getMockConfInt(tc.countConfInt)
+		mean.Result()
+		got, _ := mean.computeConfidenceIntervalSplit(0.1, 0.05) //parameters are ignored
+		if !ApproxEqual(got.LowerBound, tc.want.LowerBound) {
+			t.Errorf("TestMeanComputeConfidenceIntervalSplit(ConfidenceInterval{%f, %f})=%0.10f, want %0.10f, LowerBounds are not equal",
+				tc.meanOpt.Lower, tc.meanOpt.Upper, got.LowerBound, tc.want.LowerBound)
+		}
+		if !ApproxEqual(got.UpperBound, tc.want.UpperBound) {
+			t.Errorf("TestMeanComputeConfidenceIntervalSplit(ConfidenceInterval{%f, %f})=%0.10f, want %0.10f, UpperBounds are not equal",
+				tc.meanOpt.Lower, tc.meanOpt.Upper, got.UpperBound, tc.want.UpperBound)
+		}
+	}
+}
+
+/**
+ * This test was designed to be not deterministic. It goes along with deterministic analogues in
+ * order to ensure that they don't miss something.
+ */
+func TestMeanComputeConfidenceInterval_confidenceBoundsAlwaysInsideProvidedBoundaries(t *testing.T) {
```

Comment:
It looks like this test is randomizing anything it can get its hands on. Bt this makes it hard to understand what is happening If it throws an error and to reproduce. The test claims it wants to check that bounding is done correctly. To thest this, one deterministic test should be enough simly define a some lower bound of e.g. 0.0, upper bound of e.g. 1.0 and add interesting values -1.0, 0.0, 1.0 and 2.0